### DISCUSSION TITLE
Functions log management

### DISCUSSION THREAD

**Discussion by s3ppo** [appwrite/appwrite#2970] [Ideas]
Hi!

i wanted to start a discussion because of the log management.. it would be very cool to search for things in the logs, or maybe to remove a log after i found that this function did just nothing.

in my case i have some functions that are triggered by cron interval. lots of them did as i said before just nothing.. but it haves a log with hundreds of pages.. if i now try to help a user and want to check what happend, this is nearly impossible ;)

best regards
Harald

**Comment by Meldiron** [comment]
Here are some of my ideas, I would love to hear validation from others on them:
- Function-level setting to enable or disable logs (per specific type like HTTP, schedule, event..)
- Parameter in `createExecution` to decide if store the log or not
- Disable all logs from sync executions

**Comment by s3ppo** [comment]
a filter in the console could also be very helpful.. maybe:
- status (green or red)
- execution time (from - to)
- searchfield.. occurency of a text inside stdout

**Comment by radvansky-tomas** [comment]
My thoughts are here: https://github.com/appwrite/appwrite/issues/3295

All I am trying to explain there is that there is a big difference between result (promise outcome resolve/reject) and LOGS.

So as mentioned above:
- search
- filters
- status
- different level of error
- maybe even notifications etc
